autowarefoundation/autoware
# Modify mssion state for fms

## Status
**DEVELOPMENT**

## Description
4. rename a state and transition key of Mission state machine of autowarefoundation/autoware_ai#564 

If the parameter use_fms is true, wait at MissionInit state until `state_mission_initialized` key is given  by other node.
This should be used with service management system(e.g. [AMS](https://github.com/CPFL/AMS)).
![improve decision_maker (3)](https://user-images.githubusercontent.com/38061530/57819956-b4608d80-77c5-11e9-8b01-7d33d73d6a4c.png)

## Related Issue
autowarefoundation/autoware_ai#564 : improve decision_maker

## Todos

## Steps to Test
1. Open run time manager and click `[app]` next to `Decision -> decision_maker(experimental)` in Computing tab.
2. Check the `use_fms` checkbox and push OK button.
2. Please follow [Start Driving](https://github.com/s-azumi/Autoware/tree/feature/split_drive_state/ros/src/computing/planning/decision/packages/decision_maker#start-driving-with-simulation) until step 5.
3. Publish `/state_cmd` topic with `state_mission_initialized` key, then the state changes to `WaitOrder` from `MissionInit` state.
